### ISSUE TITLE
MLflow: results logged, confuse path with docker

### ISSUE THREAD

**Issue by davy-blavette** [ultralytics/ultralytics#17691]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

This is a strange problem, locally with the same parameters I have no error with mlflow :

```
1 - MLflow : logging run_id(3d847c6c1f66489590e04a82b73a0067) to /AI_Workspace/Server/mlflow/mlruns
2 - MLflow: see http://127.0.0.1:5000 with ‘mlflow server --backend-store-uri /AI_Workspace/Server/mlflow/mlruns’.
3 - MLflow: deactivated with ‘yolo settings mlflow=False’.

4 - Speed: 0.5ms pre-processing, 1.3ms inference, 0.0ms loss, 0.0ms post-processing per image
5 - Results saved in /AI_Workspace/Projects/C46/datasets/run58

6 - MLflow: results saved in /AI_Workspace/Server/mlflow/mlruns
7 - MLflow: deactivated with ‘yolo settings mlflow=False’.
8 - Training complete
```

but if I put my code on docker and launch via REST, this is what I get:

```
1 - MLflow : [0mlogging run_id(5d95beeaba1e49dbbc04c47a35dafd1c) to /AI_Workspace/Server/mlflow/mlruns

2 - MLflow : view at http://127.0.0.1:5000 with ‘mlflow server --backend-store-uri /AI_Workspace/Server/mlflow/mlruns’
3 - MLflow:deactivate with ‘yolo settings mlflow=False’

4 - Speed: 0.4ms pre-processing, 1.1ms inference, 0.0ms loss, 0.0ms post-processing per image
5 - Results saved in /AI_Workspace/Projects/C46/datasets/run59
```

and after I have a bug : 

```
PermissionError : [Errno 13] Permission denied : ‘/AI_Workspace/Training/mlflow/mlruns/856523503236198386/5d95beeaba1e49dbbc04c47a35dafd1c’
```

For some reason, it tries to write to /AI_Workspace/**Training**/mlflow/mlruns/ instead of /AI_Workspace/**Server**/mlflow/mlruns/.

yet in steps 1 and 2 it uses the correct path /AI_Workspace/Server/ which I configure with : 

```
MLFLOW_TRACKING_URI = os.path.join(LOCATION, 'Server', 'mlflow', 'mlruns')
os.environ[‘MLFLOW_TRACKING_URI’] = MLFLOW_TRACKING_URI,
```
 the two codes are exactly the same, just the second is mounted on docker...

### Environment

Ultralytics 8.3.27 🚀 Python-3.12.5 torch-2.4.1+cu118 CUDA:0 (Quadro P4000, 8118MiB)

### Minimal Reproducible Example

/Training/train.py (with Rest on docker)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @davy-blavette, thank you for reaching out and sharing your issue with Ultralytics 🚀! This is an automated response, but rest assured, an Ultralytics engineer will take a closer look soon.

We suggest checking out our [Docs](https://docs.ultralytics.com) for various usage examples that may already have answers to your questions.

As this is a 🐛 Bug Report, please consider providing a more detailed [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will help us to better understand and debug the issue you’re facing.

If you have questions concerning custom training configurations ❓, please include as much detail as possible, such as dataset examples and training logs. Make sure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For real-time support, feel free to drop by our [Discord](https://ultralytics.com/discord) 🎧. If you prefer detailed discussions, our [Discourse](https://community.ultralytics.com) is another excellent choice. You can also engage with knowledge sharing on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please ensure you are using the latest version of the `ultralytics` package, along with all its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Execute this in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO supports the following tested and verified environments (with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) pre-installed):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Check out the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it indicates that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests ensure the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu, running every 24 hours and after every commit.

**Comment by Y-T-G** [comment]
What's the code? If you're reporting a bug, it should be accompanied by a minimum reproducible example. And the output of `yolo checks` for the environment. These are all mentioned in the issue template.

Otherwise, it would be counted as non-reproducible.

**Comment by davy-blavette** [comment]
ok I simplifie, maybe you can reproduce with : 
```

            model = trainer_class(args, pid)
            model.train()

engine/trainer: task=classify, mode=train, model=/AI_Workspace/Training/utils/pretrained/yolov8n-cls.pt, data=/Pattyn/VisionIA/AI_Workspace/Projects/C46/datasets, epochs=2, time=None, patience=100, batch=16, imgsz=352, save=True, save_period=-1, cache=False, device=None, workers=8, project=C46, name=run58, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=/AI_Workspace/Projects/C46/datasets/run58

settings.update({'weights_dir': PRETRAINED_PATH})
settings.update({'wandb': False})

class Yolov8Trainer:
    def __init__(self, args, pid):

        self.pid = pid
        self.file_handler = FileHandler(self.pid)

        self.project = self.file_handler.project
        self.datasets = self.file_handler.dataset
        self.args_file = self.file_handler.args_file
        self.data = self.file_handler.dataset if pid[0] == "C" else self.file_handler.data_file

        self.pretrained_model = args.pretrained
        self.imgsz = int(args.imgsz)
        self.batch = int(args.batch)
        self.single_cls = args.single_cls
        self.epochs = args.epochs
        self.pid = pid

        self.model = self.load_model()
        self.configure_mlflow()


    def load_model(self):
        return YOLO(Path(f'{PRETRAINED_PATH}/{self.pretrained_model}'))

    def configure_mlflow(self):
        os.environ['MLFLOW_TRACKING_URI'] = MLFLOW_TRACKING_URI
        os.environ['MLFLOW_RUN'] = self.pid
        settings.update({'mlflow': True})

    def train(self):
        rect = self.determine_rectangular()
        train_params = self.get_train_params(rect)
        results = self.model.train(**train_params)


    def get_train_params(self, rect):
        return {
            'data': self.data,
            'epochs': self.epochs,
            'imgsz': self.imgsz,
            'batch': self.batch,
            'single_cls': self.single_cls,
            'project': self.pid,
            'name': self.file_handler.output_run,
            'rect': rect
        }
```

**Comment by Y-T-G** [comment]
I don't see how this is `ultralytics` related.

You can try setting `MLFLOW_BACKEND_STORE_URI`.

**Comment by davy-blavette** [comment]
it was a problem with old MLFLOW data which was indeed pointing to this old file, I have cleaned up my MLFLOW data and I no longer have this problem, it was not an ultralytics problem. 👍

**Comment by glenn-jocher** [comment]
Thank you for the update. I'm glad to hear the issue is resolved. If you have any further questions or need assistance with Ultralytics, feel free to ask.
